airflow/jobs/backfill_job.py

```diff
@@ -627,6 +629,11 @@ def to_keep(key: TaskInstanceKey) -> bool:
                 for new_ti in new_mapped_tis:
                     new_ti.set_state(TaskInstanceState.SCHEDULED, session=session)
 
+            # Set state to failed for running TIs that are set up for retry if disable-retry flag is set
+            for ti in list(ti_status.running.values()):
```

Comment:
Why the `list()` call?

Reply:
You are right, it's not needed. Removed.